src/Container.php

```diff
@@ -161,50 +162,38 @@ private function resolveParams(string $qn, ReflectionMethod $constructor): itera
                 yield $param->getDefaultValue();
                 continue;
             }
+
             $paramType = $param->getType();
+
             if ($paramType instanceof ReflectionUnionType) {
-                $msgFormat = 'Unable to resolve %s constructor parameter $%s (position %d): it has union type %s.';
-                $types = [];
-                foreach ($paramType->getTypes() as $namedType) {
-                    $types[] = $namedType->getName();
-                }
-                $message = sprintf(
-                    $msgFormat,
-                    $qn,
-                    $param->getName(),
-                    $param->getPosition() + 1,
-                    implode('|', $types)
-                );
-                throw new ContainerException($this->buildExceptionReport($message));
+                $this->throwNonResolvableUnionParameterTypeException($qn, $param);
             }
 
-            try {
-                /** @var ReflectionNamedType $paramType */
-                $paramClass = !$paramType || $paramType->isBuiltin()
-                    ? null
-                    : new ReflectionClass($paramType->getName());
-            } catch (ReflectionException $e) {
-                $msg = sprintf(
-                    'Unable to resolve %s constructor parameter $%s: %s.',
-                    $qn,
-                    $param->getName(),
-                    $e->getMessage()
-                );
-                throw new ContainerException($this->buildExceptionReport($msg), 0, $e);
+            if ($paramType === null) {
```

Comment:
Same as above. We can create something like `$this->checkForNull()` that will check for null and throw an exception.

Reply:
I couldn't make it work for null type checking in another method. I guess it's expected behavior and you need to define it as an extension: https://phpstan.org/developing-extensions/type-specifying-extensions

Too much overhead for simple stuff. So, I just kept it as it's.